python/GafferOSLTest/OSLExpressionEngineTest.py

```diff
@@ -888,5 +888,18 @@ def testParseErrorLineNumbers( self ) :
 		with self.assertRaisesRegex( RuntimeError, "expression:1: error: 'undefinedVariable' was not declared in this scope" ) :
 			s["expression"].setExpression( "parent.node.op1 = undefinedVariable;", "OSL" )
 
+	def testColonInPlugNames( self ) :
+
+		s = Gaffer.ScriptNode()
+		s["n"] = Gaffer.Node()
+		s["n"]["user"]["test:i"] = Gaffer.IntPlug( flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic )
+		s["n"]["user"]["test:o"] = Gaffer.IntPlug( flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic )
+
+		s["e"] = Gaffer.Expression()
+		s["e"].setExpression( "parent.n.user.test:o = parent.n.user.test:i + 1;", "OSL" )
```

Comment:
It's a bit strange that `:` isn't valid OSL syntax, but I can't think of any better options right now. The _real_ solution seems to be to stop expressions referencing foreign plugs directly, and instead have them always reference local plugs (where we could reasonably refuse to support `:` for OSL). We've talked about that for years now without making any progress - should we try to get it into 1.6, or is that biting off more than we want to chew?

Reply:
> The real solution seems to be to stop expressions referencing foreign plugs directly, and instead have them always reference local plugs (where we could reasonably refuse to support : for OSL).

After talking, we're not convinced that we'd refuse ":" in a local plug name for OSL as that may appear a little inelegant when creating an OSL expression from an existing plug with ":" in its name. So we'd likely need this fix either way...

> We've talked about that for years now without making any progress - should we try to get it into 1.6, or is that biting off more than we want to chew?

I've put it back on the radar and tentatively tagged with with 1.6 so we can at least keep it in mind and consider it a bit more seriously while 1.6 shapes up.